src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -92,6 +108,26 @@ def get(self, request: Request, project) -> Response:
                     config, filepath, config.default_branch, commit_id
                 )
 
+                if not link and frame:
+                    frame["filename"] = filepath
+                    munged_frames = munged_filename_and_frames(
+                        platform, [frame], "munged_filename", sdk_name=sdk_name
+                    )
+                    if munged_frames:
+                        munged_frame: Mapping[str, Any] = munged_frames[1][0]
+                        munged_filename = str(munged_frame.get("munged_filename"))
+                        if munged_filename:
+                            if not filepath.startswith(
+                                config.stack_root
+                            ) and not munged_filename.startswith(config.stack_root):
+                                scope.set_tag("stacktrace_link.error", "stack_root_mismatch")
+                                result["error"] = "stack_root_mismatch"
+                                continue
```

Comment:
Only return `stack_root_mismatch` when both `file` and `munged_filename` don't startwith the `config.stack_root`.

Otherwise, we attempt to do the repo-linking on the `munged_filename` and that should either succeed or result in `file_not_found`